shotover-proxy/src/transforms/query_counter.rs

```diff
@@ -28,12 +29,19 @@ impl QueryCounter {
 
 #[async_trait]
 impl Transform for QueryCounter {
-    async fn transform<'a>(&'a mut self, message_wrapper: Wrapper<'a>) -> ChainResponse {
-        for m in &message_wrapper.messages {
-            if let MessageDetails::Query(QueryMessage { ast: Some(ast), .. }) = &m.details {
-                match ast {
-                    ASTHolder::SQL(statement) => {
-                        let query_type = match **statement {
+    async fn transform<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
+        for m in &mut message_wrapper.messages {
+            match &m.original {
+                Frame::Cassandra(CassandraFrame {
+                    operation:
+                        CassandraOperation::Query {
+                            query: CQL::Parsed(query),
+                            ..
+                        },
+                    ..
+                }) => {
+                    for statement in query {
+                        let query_type = match statement {
                             Statement::Query(_) => "SELECT",
```

Comment:
Why not use something like the `QueryType` enum? Do we want this metric to be specific or more generic (i.e. "write" or "read")?

Reply:
I think being more specific is useful.
But theres nothing stopping us from from having a QueryType metric in another transform.
You can raise an issue if you want to make a design discussion about the kinds of metrics we report.